hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -72,11 +72,49 @@ func TunedConfigMap(namespace, name string) *corev1.ConfigMap {
 	}
 }
 
-func PerformanceProfileConfigMap(namespace, name string) *corev1.ConfigMap {
+func PerformanceProfileConfigMap(namespace, name, nodePoolName string) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: namespace,
-			Name:      "perfprofile-" + name,
+			Name:      ComposeValidName(name, nodePoolName),
 		},
 	}
 }
+
+// 63 is the qualifiedNameMaxLength allowed for k8s object
+// https://github.com/kubernetes/kubernetes/blob/957c9538670b5f7ead2c9ba9ceb9de081d66caa4/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L34
+const qualifiedNameMaxLength = 63
+
+// ComposeValidName takes two qualified objects` names
+func ComposeValidName(name1, name2 string) string {
```

Comment:
@csrwng thank you for the review.
I understand your concern, but the shortenName function might returns an unreadable name like:
In my case I want the `PerformanceProfile` to be compose of The `ConfigMap` name + `NodePool` name
If i'll use the shortenName function, in some edge cases I might be getting something like:
[hash]-[NodePool name] 

While with this function we have at least some weight prefixing.

OTOH, your argument in the comment is completely right. 
This  PR became much more complex than I anticipated and I would close it since it doesn't worth the effort anymore, but the problem is that the existing implementation has a bug as well:
https://github.com/openshift/hypershift/pull/4108#discussion_r1627581711
> BTW the previous implementation was problematic as well, because we are not checking the length of "perfprofile-+name"